addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlBridgeHandler.java

```diff
@@ -0,0 +1,266 @@
+     *
+     * @param addr IP address of Niko Home Control IP-interface or null
+     * @param port
+     */
+    private void createCommunicationObject(InetAddress addr, int port, InetAddress broadcastAddr) {
+
+        Configuration config = this.getConfig();
+
+        scheduler.submit(new Runnable() {
+
+            @Override
+            public void run() {
+                try {
+                    nhcComm = new NikoHomeControlCommunication(addr, broadcastAddr);
+
+                    nhcComm.startCommunication(port);
+                    if (!nhcComm.communicationActive()) {
+                        throw new IOException("Niko Home Control: communication socket error");
+                    }
+
+                    // Set callback from NikoHomeControlCommunication object to this bridge to be able to take bridge
+                    // offline when non-resolvable communication error occurs.
+                    setBridgeCallBack();
+
+                    updateProperties();
+
+                    updateStatus(ThingStatus.ONLINE);
+
+                    Integer refreshInterval = ((Number) config.get(CONFIG_REFRESH)).intValue();
+                    setupRefreshTimer(refreshInterval);
+
+                    if (nhcDiscovery != null) {
+                        nhcDiscovery.discoverDevices();
+                    } else {
+                        logger.debug("Niko Home Control: cannot discover, discovery service not started");
+                    }
+
+                } catch (IOException e) {
+                    nhcComm = null;
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                            "Niko Home Control: error starting bridge connection");
+                }
+            }
+
+        });
+
+    }
+
+    private void setBridgeCallBack() {
+        this.nhcComm.setBridgeCallBack(this);
+    }
+
+    /**
+     * Schedule future communication refresh.
+     *
+     * @param interval_config Time before refresh in minutes.
+     */
+    private void setupRefreshTimer(Integer refreshInterval) {
+
+        if (this.refreshTimer != null) {
+            this.refreshTimer.cancel(true);
+            this.refreshTimer = null;
+        }
+
+        if ((refreshInterval == null) || (refreshInterval == 0)) {
+            return;
+        }
+
+        // This timer will restart the bridge connection periodically
+        logger.debug("Niko Home Control: restart bridge connection every {} min", refreshInterval);
+        this.refreshTimer = scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+
+                logger.debug("Niko Home Control: restart communication at scheduled time");
+
+                try {
+                    nhcComm.restartCommunication();
+                    if (!nhcComm.communicationActive()) {
+                        throw new IOException("Niko Home Control: communication socket error");
```

Comment:
Thank you for the advice. I will keep it in mind.
I think this is a remain from previous code where restartCommunication() could also throw the IOException. I took it out.